sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -723,7 +723,7 @@ class SQLContext private[sql](val sparkSession: SparkSession)
    * @since 1.3.0
    */
   def tables(databaseName: String): DataFrame = {
-    Dataset.ofRows(sparkSession, ShowTablesCommand(Some(databaseName), None))
+    Dataset.ofRows(sparkSession, ShowTablesCommand(Option(databaseName), None))
```

Comment:
CC maybe @yhuai - OK to let `tables(null)` act like `tables()`? seems reasonable